lib/nodejs/lib/thrift/connection.js

```diff
@@ -198,6 +200,16 @@ Connection.prototype.connection_gone = function () {
   var self = this;
   this.connected = false;
 
+  // If closed by manual, emit close event and cancel reconnect process
+  if(this.forceClose) {
+    self.emit("close");
```

Comment:
This may be overly pedantic, but would it be better to *first* clear the timer and *then* emit close? In other words, move the line `self.emit("close");` down below the `if (this.retry_timer) {` block?

Reply:
> 

Correct, `close callback` maybe cause pedantic.
I have updated this change